spikes/npgsql-vanilla/DemoProgram.cs

```diff
@@ -0,0 +1,176 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using CommandLine;
+using Npgsql;
+using NpgsqlTypes;
+
+namespace demo
+{
+    public static class DemoProgram
+    {
+        private static async Task SystemQueryExample(NpgsqlConnection conn)
+        {
+            Console.WriteLine("Running SystemQueryExample");
+            await using (var cmd = new NpgsqlCommand("SELECT mountain FROM sys.summits ORDER BY 1 LIMIT 25", conn))
+            await using (var reader = cmd.ExecuteReader())
+            {
+                var mountains = new List<string>();
+                while (reader.Read())
+                {
+                    mountains.Add(reader.GetString(0));
+                }
+
+                Console.WriteLine($"Mountains: {string.Join(",", mountains)}");
+            }
+
+            Console.WriteLine();
+        }
+
+        private static async Task BasicConversationExample(NpgsqlConnection conn)
+        {
+            Console.WriteLine("Running BasicConversationExample");
+            
+            // Submit DDL, create database schema.
+            await using (var cmd = new NpgsqlCommand("DROP TABLE IF EXISTS testdrive.basic", conn))
+            {
+                cmd.ExecuteNonQuery();
+            }
+            await using (var cmd = new NpgsqlCommand("CREATE TABLE testdrive.basic (x int)", conn))
+            {
+                cmd.ExecuteNonQuery();
+            }
+
+            // Insert single data point.
+            using (var cmd = new NpgsqlCommand("INSERT INTO testdrive.basic (x) VALUES (@x)", conn))
+            {
+                cmd.Parameters.AddWithValue("x", -999);
+                cmd.ExecuteNonQuery();
+            }
+
+            // Insert multiple data points.
+            await using (var cmd = new NpgsqlCommand("INSERT INTO testdrive.basic (x) VALUES (@x)", conn))
+            {
+                await using (var transaction = conn.BeginTransaction())
```

Comment:
Should we even include transactions (basically not supported in CrateDB)

Reply:
I believe this is here to demonstrate that this will act as a no-op instead of croaking.

Reply:
Ah ... now I get it ... I am wondering if we really want to use the full test for a simple example.
Considering also that the whole connection part is on the bottom :/

Reply:
Afaik using the transaction is important to get efficient bulk insert execution. But would need to double check the message flow.